File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        for(int i=0;i<s.size();i++)
        {
            if(!st.empty()){
                if((s[i]==')' && st.top()=='(') || (s[i]==']' && st.top()=='[') ||  (s[i]=='}' && st.top()=='{')){
                    st.pop();
                    continue;
                }
            } 
            st.push(s[i]);
        }
        if(!st.empty())
        {
            return false;
        }
        return true;
        
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack <Character> st=new Stack<Character>();
        for(char c:s.toCharArray())
        {
            if(c=='(')
            {
                st.push(')');
                
            }
            else if(c=='{')
            {
                st.push('}');
            }
            else if(c=='[')
            {
                st.push(']');
            }
            else if(st.isEmpty()||st.pop()!=c)
            {
                return false;
            }
        }
        
        return st.isEmpty();
        
    }
}





// public boolean isValid(String s) {
// 	Stack<Character> stack = new Stack<Character>();
// 	for (char c : s.toCharArray()) {
// 		if (c == '(')
// 			stack.push(')');
// 		else if (c == '{')
// 			stack.push('}');
// 		else if (c == '[')
// 			stack.push(']');
// 		else if (stack.isEmpty() || stack.pop() != c)
// 			return false;
// 	}
// 	return stack.isEmpty();
// }